src/test/java/teammates/test/cases/ui/pagedata/InstructorFeedbackResultsPageDataTest.java

```diff
@@ -0,0 +1,1584 @@
+import java.util.Set;
+
+import org.testng.annotations.BeforeClass;
+import org.testng.annotations.Test;
+
+import teammates.ui.controller.InstructorFeedbackResultsPageData.ViewType;
+import teammates.common.datatransfer.AccountAttributes;
+import teammates.common.datatransfer.DataBundle;
+import teammates.common.datatransfer.FeedbackQuestionAttributes;
+import teammates.common.datatransfer.FeedbackResponseAttributes;
+import teammates.common.datatransfer.InstructorAttributes;
+import teammates.common.datatransfer.StudentAttributes;
+import teammates.common.exception.EntityDoesNotExistException;
+import teammates.common.exception.UnauthorizedAccessException;
+import teammates.common.util.Const;
+import teammates.common.util.Sanitizer;
+import teammates.common.util.StringHelper;
+import teammates.logic.api.Logic;
+import teammates.test.cases.BaseComponentTestCase;
+import teammates.ui.controller.InstructorFeedbackResultsPageData;
+import teammates.ui.template.ElementTag;
+import teammates.ui.template.InstructorFeedbackResultsGroupByParticipantPanel;
+import teammates.ui.template.InstructorFeedbackResultsGroupByQuestionPanel;
+import teammates.ui.template.InstructorFeedbackResultsModerationButton;
+import teammates.ui.template.InstructorFeedbackResultsResponsePanel;
+import teammates.ui.template.InstructorFeedbackResultsResponseRow;
+import teammates.ui.template.InstructorFeedbackResultsSecondaryParticipantPanelBody;
+import teammates.ui.template.InstructorFeedbackResultsSectionPanel;
+import teammates.ui.template.InstructorFeedbackResultsParticipantPanel;
+import teammates.ui.template.InstructorFeedbackResultsQuestionTable;
+
+public class InstructorFeedbackResultsPageDataTest extends BaseComponentTestCase {
+    private static final String NOT_IN_A_SECTION = "Not in a section";
+
+    private static DataBundle dataBundle = getTypicalDataBundle();
+    
+    // logic is used in this page data test, but not other page data tests, because
+    // unfortunately, logic is required to construct the ResultsBundle
+    Logic logic = new Logic();
+    
+    @BeforeClass
+    public static void classSetUp() throws Exception {
+        printTestClassHeader();
+        removeAndRestoreTypicalDataInDatastore();
+    }
+    
+    @Test
+    public void testInitForViewByQuestion() throws UnauthorizedAccessException, EntityDoesNotExistException {
+        AccountAttributes account = dataBundle.accounts.get("instructor1OfCourse1");
+        InstructorFeedbackResultsPageData data = new InstructorFeedbackResultsPageData(account);
+        
+        InstructorAttributes instructor = dataBundle.instructors.get("instructor1OfCourse1");
+        
+        ______TS("typical case: view all sections, all questions, show stats");
+        
+        String courseId = instructor.courseId;
+        String feedbackSessionName = dataBundle.feedbackSessions.get("session1InCourse1").feedbackSessionName;
+        String showStats = "on";
+        String groupByTeam = "on";
```

Comment:
Should "on" be made into a constant for easier tracing later on when they're changed to booleans?